Realm/ObjectServerTests/RLMPermissionsAPITests.m

```diff
@@ -804,4 +802,103 @@ - (void)testSettingUnownedRealmPermission {
     CHECK_PERMISSION_ABSENT(results, expectedPermission);
 }
 
+#pragma mark - Delete Realm upon permission denied
+
+/// A Realm which is opened improperly should report an error allowing the app to recover.
+- (void)testDeleteRealmUponPermissionDenied {
+    __block void(^workBlock)(NSError *, NSURL *url) = ^(NSError *err, __unused NSURL *url) {
+        XCTFail(@"Error handler should not be called unless explicitly expected. Error: %@", err);
+    };
+    [[RLMSyncManager sharedManager] setErrorHandler:^(NSError *error, RLMSyncSession *session) {
```

Comment:
Why are these tests mutating a local variable captured by this block rather than just changing the errorHandler block?